### PR TITLE
Update stack and cabal dependencies

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: .stack-to-nix.cache [hunk 1]
```diff
@@ -146,3 +146,6 @@ https://github.com/input-output-hk/cardano-ledger ccb81fc69285c127c47ce622c49963
 https://github.com/input-output-hk/cardano-ledger ccb81fc69285c127c47ce622c4996317556aff42 cardano-ledger/test 0ayq8f7dgdgwbgjn7wvh7m94sa7nbgm5633hm62z3frp02l2zgfl cardano-ledger-test cardano-ledger-test.nix
 https://github.com/input-output-hk/cardano-ledger ccb81fc69285c127c47ce622c4996317556aff42 crypto 0ayq8f7dgdgwbgjn7wvh7m94sa7nbgm5633hm62z3frp02l2zgfl cardano-crypto-wrapper cardano-crypto-wrapper.nix
 https://github.com/input-output-hk/cardano-ledger ccb81fc69285c127c47ce622c4996317556aff42 crypto/test 0ayq8f7dgdgwbgjn7wvh7m94sa7nbgm5633hm62z3frp02l2zgfl cardano-crypto-test cardano-crypto-test.nix
+https://github.com/input-output-hk/iohk-monitoring-framework 9cb21a3ebf0f45bfb94ece3e6d62e613eca8b17b iohk-monitoring 1nms4gap3j6qc0gs4rk8kpqm6zdqcp5hchsd1wv459wvhfq4f20p iohk-monitoring iohk-monitoring.nix
```
Comment (mrBliss) [inline .stack-to-nix.cache]: Do we even need to store `.stack-to-nix.cache` in git?

Comment (erikd) [reply .stack-to-nix.cache]: I plead complete ignorance for all the Nix stuff.

I my opinion, there should be **no autogenerated files at all** committed to git, **ever**.

**Comment by erikd** [comment]
@avieth @mrBliss does this not suggest that `byron-proxy` should be in a separate repsitory that could pin specific versions of all its dependencies?

File: .stack-to-nix.cache (hunk 1 above)
Comment (avieth) [reply .stack-to-nix.cache]: I'll bet we need it for the benefit of CI.

**Comment by avieth** [comment]
> does this not suggest that byron-proxy should be in a separate repsitory that could pin specific versions of all its dependencies?

I have no objection to that.

> What about stack.byron-proxy.yaml and byron-proxy/cabal.project? Is the former even necessary? Are we not keeping the latter in sync with the top-level cabal.project?

I use nix to build, and so does CI. AFAIK the nix derivation generation script only sources the stack.yaml to figure out the dependencies, but I also know that the IOHK nix tools are capable of using cabal.project, although in a roundabout way by generating a plan.json first then using that.

**Pull request merged**